flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -95,6 +103,8 @@ public class IcebergSource<T> implements Source<T, IcebergSourceSplit, IcebergEn
   private final SerializableRecordEmitter<T> emitter;
   private final String tableName;
 
+  private volatile List<IcebergSourceSplit> batchSplits;
```

Comment:
is this called in multithreaded way?

Reply:
`planSplitsForBatch` can be called twice: (1) source/stream construction by main thread (2) enumerator creation.

confirmed via debugger they are done by different threads.

(1) from main thread
<img width="575" alt="image" src="https://github.com/user-attachments/assets/eac07c33-8ee3-4648-ac2d-2812ac0dcb44">

(2) from actor thread
<img width="633" alt="image" src="https://github.com/user-attachments/assets/1e296e28-1f5f-4465-8685-4f128cca2f2d">